AssetsView/Winforms/TextureViewer.cs

```diff
@@ -124,6 +126,31 @@ private void TextureViewer_FormClosed(object sender, FormClosedEventArgs e)
             }
         }
 
+        public void SaveTexture()
+        {
+            var res = MessageBox.Show("Would you like to save the texture as .PNG?", "Texture", MessageBoxButtons.YesNo);
+            if (res != DialogResult.Yes)
+            {
+                return;
+            }
+
+            SaveFileDialog tex_save_dialog = new SaveFileDialog();
+            tex_save_dialog.Filter = ".PNG File|*.png";
+            tex_save_dialog.Title = "Save texture as .PNG file";
+            res = tex_save_dialog.ShowDialog();
+            string selected_name = tex_save_dialog.FileName;
+            if (res == DialogResult.Cancel || res == DialogResult.No || string.IsNullOrWhiteSpace(selected_name))
+            {
+                return;
+            }
+            if (File.Exists(selected_name))
+            {
+                File.Delete(selected_name);
+            }
+            image.Save(selected_name, ImageFormat.Png);
```

Comment:
I would also suggest adding a null check since image is null if the image failed to load.